src/NUnitCore/core/NUnitFramework.cs

```diff
@@ -100,12 +100,15 @@ internal static Assembly FrameworkAssembly
             {
                 if (!frameworkAssemblyInitialized)
                     foreach (Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
-                        if (assembly.GetName().Name == "nunit.framework" ||
-                            assembly.GetName().Name == "NUnitLite")
+                    {
+                        AssemblyName name = assembly.GetName();
+                        if (name.Name == "nunit.framework" && name.Version.Major == 2 ||
+                            name.Name == "NUnitLite" && name.Version.Major == 1)
```

Comment:
This improves the code but leaves a problem: If you force two different versions of NUnit V2 to run in the same AppDomain, you'll get the wrong version for one assembly. However, that's a corner case on a corner case and I'm happy to let it go!